internal/db/user_resolve.go

```diff
@@ -9,14 +9,62 @@ import (
 	"github.com/edulinq/autograder/internal/model"
 )
 
+// Returns a sorted list of users based on the course reference and an error.
+func ResolveCourseUsers(course *model.Course, reference *model.CourseUserReference) ([]*model.CourseUser, error) {
```

Comment:
I think both the name and semantics here need to be changed slightly,
since one of my previous comments said that outside emails should probably be allowed.